internal/pkg/skuba/addons/addons.go

```diff
@@ -252,11 +261,61 @@ func (addon Addon) Apply(client clientset.Interface, addonConfiguration AddonCon
 		}
 		renderedManifest = string(renderedManifestBytes)
 	}
-	cmd := exec.Command("kubectl", "apply", "--kubeconfig", skubaconstants.KubeConfigAdminFile(), "-f", "-")
-	cmd.Stdin = bytes.NewBuffer([]byte(renderedManifest))
-	if combinedOutput, err := cmd.CombinedOutput(); err != nil {
-		klog.Errorf("failed to run kubectl apply: %s", combinedOutput)
-		return err
+
+	var obj unstructured.Unstructured
+	yamlDecoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(renderedManifest)), 4096)
+	for { // Process each document in the yaml stream
+		decodeErr := yamlDecoder.Decode(&obj)
+		if decodeErr == io.EOF {
+			break
+		}
+		data, err := runtime.Encode(unstructured.UnstructuredJSONScheme, &obj)
+		if err != nil {
+			return err
+		}
+		accessor := meta.NewAccessor()
+		name, err := accessor.Name(&obj)
+		if err != nil {
+			return err
+		}
+		namespace, err := accessor.Namespace(&obj)
+		if err != nil {
+			return err
+		}
+		forceConflicts := true
+		options := &metav1.PatchOptions{
+			Force:        &forceConflicts,
+			FieldManager: "skuba",
+		}
+		gvk := obj.GroupVersionKind()
+		gk := schema.GroupKind{Group: gvk.Group, Kind: gvk.Kind}
+		groupResources, err := restmapper.GetAPIGroupResources(client.Discovery())
+		if err != nil {
+			return err
+		}
+		mapping, err := restmapper.NewDiscoveryRESTMapper(groupResources).RESTMapping(gk, gvk.Version)
+		if err != nil {
+			return err
+		}
+		cfg, err := ioutil.ReadFile(skubaconstants.KubeConfigAdminFile())
+		if err != nil {
+			return err
+		}
+		clientCfg, err := clientcmd.RESTConfigFromKubeConfig(cfg)
+		if err != nil {
+			return err
+		}
+		dynamicClient, err := dynamic.NewForConfig(clientCfg)
+		if err != nil {
+			return err
+		}
+		_, err = dynamicClient.Resource(mapping.Resource).
+			Namespace(namespace).
+			Patch(name, "application/apply-patch+yaml", data, *options)
```

Comment:
```suggestion
			Patch(name, types.ApplyPatchType, data, *options)
```

where package `types` is `k8s.io/apimachinery/pkg/types`.